### PR TITLE
Add CMake export

### LINKED ISSUE #30: Add cmake export of proteus
We need a cmake to export proteus for integration to user software

### PR DESCRIPTION
**Pull request by davidbeckingsale**
A user can `find_package(proteus)` and then depend on the targets `proteus` and `ProteusPass` to automatically get the correct compile and link flags.

Fixes #30

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -40,3 +40,25 @@ if(ENABLE_TESTS)
...
+  COMPATIBILITY AnyNewerVersion)
+
+install(EXPORT proteusTargets
+  DESTINATION lib/cmake/proteus)
+
+install(FILES
+  "${PROJECT_BINARY_DIR}/proteusConfigVersion.cmake"
+  "${PROJECT_BINARY_DIR}/proteusConfig.cmake"
+  DESTINATION
+  lib/cmake/proteus)
```
Comment (davidbeckingsale) [inline CMakeLists.txt:64]:
```suggestion
  lib/cmake/proteus)
  
```

File: cmake/proteusConfig.cmake.in [hunk 2]
```diff
@@ -0,0 +1,6 @@
+@PACKAGE_INIT@
+
+find_package(LLVM 17 REQUIRED CONFIG)
+
+include("${CMAKE_CURRENT_LIST_DIR}/proteusTargets.cmake")
+check_required_components("@PROJECT_NAME@")
```
Comment (davidbeckingsale) [inline cmake/proteusConfig.cmake.in:6]:
```suggestion
check_required_components("@PROJECT_NAME@")

```

File: lib/CMakeLists.txt [hunk 3]
```diff
@@ -64,8 +64,13 @@ if(ENABLE_CUDA)
...
 endif()
 
-target_link_libraries(proteusjit PRIVATE ${libs} "$<$<PLATFORM_ID:Darwin>:-undefined dynamic_lookup>")
-set_target_properties(proteusjit PROPERTIES
+target_link_libraries(proteus PRIVATE ${libs} "$<$<PLATFORM_ID:Darwin>:-undefined dynamic_lookup>")
+set_target_properties(proteus PROPERTIES
   INSTALL_RPATH_USE_LINK_PATH TRUE)
 
-install(TARGETS proteusjit)
+install(
```
Comment (ggeorgakoudis) [inline lib/CMakeLists.txt:71]: Add also header libraries for JitEngine* as the API to clients

File: pass/CMakeLists.txt [hunk 4]
```diff
@@ -1,15 +1,26 @@
-add_library(ProteusJitPass SHARED ProteusJitPass.cpp)
+add_library(ProteusPass SHARED ProteusJitPass.cpp)
```
Comment (ggeorgakoudis) [inline pass/CMakeLists.txt:1]: nit: Do we also want to rename the source file to `ProteusPass.cpp`?

**Review by ggeorgakoudis — CHANGES_REQUESTED** [review]
Looks good, thank you! Few minor comments.

**Pull request merged**